### PR TITLE
Restore java 8 support

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
~CI is running into an error reading~

```
unsupported Java version: 11
```

~Using a newer jruby seems to fix it.~

EDIT: I'm reusing this PR with a different intention from the original one. Some commits present in the original version of this PR (which intended to add Java 11 support) were introduced directly to the master branch, and some of them broke Java 8 support. This PR now restores Java 8 support, and also brings the additional changes resulting from rebasing the original changes against the new master branch.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
Still a WIP since I want to work on some stuff that I skipped when I shouldn't, but I have a green baseline now supportting jruby 9.2 and java 11: https://github.com/deivid-rodriguez/jruby-maven-plugins/actions/runs/964460637.

**Comment by deivid-rodriguez** [comment]
Hei @mkristian, I see that you pushed some of the patches that used to live in this WIP PR to master and released 2.0.0. I was just trying things and wasn't really expecting anything to be merged as it was, but I'm glad things are moving forward.

Anyways, I was working on fixing CI for https://github.com/jruby/ruby-maven/pull/4, since we need it (or at least it'd be desirable) to be able to release a new version. However, we're running into a rubygems errors due to `polyglot-ruby` using a version of `gem-maven-plugin` that did not include https://github.com/torquebox/jruby-maven-plugins/pull/106, which was released with 2.0.0.

So I head over polyglot-ruby to get the version of `gem-maven-plugin` they use upgraded and opened https://github.com/takari/polyglot-maven/pull/236. However, when testing locally I run into the following error:

```
[ERROR] Failed to execute goal de.saumya.mojo:gem-maven-plugin:2.0.0-1:initialize (default) on project polyglot-ruby: Execution default of goal de.saumya.mojo:gem-maven-plugin:2.0.0-1:initialize failed: Unable to load the mojo 'initialize' in the plugin 'de.saumya.mojo:gem-maven-plugin:2.0.0-1' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: de/saumya/mojo/gem/InitializeMojo has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

That made me realize that `jruby-maven-plugins` unintentionally dropped support for Java 8. So I came back here and rebased this PR, restoring Java 8 support. With this PR in place, https://github.com/takari/polyglot-maven/pull/236 is almost green (one single failure which I would need to investigate).

Can you review this and let me know if it makes sense to you?

File: parent-mojo/pom.xml [hunk 1]
```diff
@@ -108,8 +108,8 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
           <configuration>
-            <source>11</source>
-            <target>11</target>
+            <source>8</source>
+            <target>8</target>
```
Comment (deivid-rodriguez) [inline parent-mojo/pom.xml:112]: This, and the equivalent change in the root pom file, is the specific change that restores support. Java 8 should be used for compiling as long as we want to keep Java 8 support.

**Review by mkristian — APPROVED** [review]
lgtm

**Comment by deivid-rodriguez** [comment]
Awesome @mkristian, thanks so much! Looking forward for a new release including this 💪.

**Comment by mkristian** [comment]
let me release this today . . .

**Pull request merged**